### PR TITLE
MGMT-16410: OpenAPI spec - added resource endpoints

### PR DESCRIPTION
**Pull request by danielerez**
* Added resource endpoints (resourceTypes/resourcePools/resources) to the OpenAPI spec.
* Added a makefile target for validting the spec.yaml file.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,14 +51,19 @@ lint:
 	golangci-lint --version
 	golangci-lint run --verbose --print-resources-usage --modules-download-mode=vendor --timeout=5m0s
 
+.PHONY: validate-openapi
+validate-openapi:
+	@echo "Run OpenAPI validator"
+	go run github.com/getkin/kin-openapi/cmd/validate@v0.119.0 internal/openapi/spec.yaml
```
Comment (jhernand) [inline Makefile:57]: There is already a test that runs the `spectral` tool:

https://github.com/openshift-kni/oran-o2ims/blob/5daf9b9676c32c5248a448f5edd41c45f32735a6/internal/openapi/handler_test.go#L83-L117

If you still prefer to have this Makefile target I won't object, but it is a bit redundant. If you prefer this please remove that test.

**Review by jhernand — COMMENTED** [review]
Looks good, but you may want to address the comment about using both "spectral" and "kin-openapi".

File: Makefile (hunk 1 above)
Comment (danielerez) [reply Makefile:57]: Oh missed this test, indeed looks good enough. I'll remove the target.

**Pull request merged**